osu.Game/Rulesets/Objects/PathControlPoint.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets.Objects.Types;
+using osuTK;
+
+namespace osu.Game.Rulesets.Objects
+{
+    public class PathControlPoint : IEquatable<PathControlPoint>
+    {
+        /// <summary>
+        /// The position of this <see cref="PathControlPoint"/>.
+        /// </summary>
+        public readonly Bindable<Vector2> Position = new Bindable<Vector2>();
+
+        /// <summary>
+        /// The type of path segment starting at this <see cref="PathControlPoint"/>.
+        /// If null, this <see cref="PathControlPoint"/> will be a part of the previous path segment.
+        /// </summary>
+        public readonly Bindable<PathType?> Type = new Bindable<PathType?>();
+
+        /// <summary>
+        /// Invoked when any property of this <see cref="PathControlPoint"/> is changed.
+        /// </summary>
+        internal event Action Changed;
+
+        /// <summary>
+        /// Creates a new <see cref="PathControlPoint"/>.
+        /// </summary>
+        public PathControlPoint()
```

Comment:
i'd usually do this the other way around (have the `ctor` with parameters call the one without) since you're initialising with already-initialised values.

Reply:
I'm confused, this is doing exactly what you want?

Reply:
```
        /// <summary>
        /// Creates a new <see cref="PathControlPoint"/> with a provided position and type.
        /// </summary>
        /// <param name="position">The initial position.</param>
        /// <param name="type">The initial type.</param>
        public PathControlPoint(Vector2 position, PathType? type = null) : this()
        {
            Position.Value = position;
            Type.Value = type;
        }


/// <summary>
        /// Creates a new <see cref="PathControlPoint"/>.
        /// </summary>
        public PathControlPoint()
{
            Position.ValueChanged += _ => Changed?.Invoke();
            Type.ValueChanged += _ => Changed?.Invoke();
}
```

Reply:
Oh I misread, thought this comment was inside SliderPath... >_<